Packs/DigitalGuardian/Integrations/DigitalGuardianARCEventCollector/DigitalGuardianARCEventCollector.py

```diff
@@ -18,276 +17,286 @@
+    Raises:
+        DemistoException | HTTPError: If request failed.
     """
-    try:
-        limit = arg_to_number(params.get('number_of_events')) or 1000
-        fetch_events(
-            client=client,
-            last_run={},
-            limit=limit,
-        )
+    invalid_export_profiles = set()
 
-    except Exception as e:
-        if 'Forbidden' in str(e):
-            raise DemistoException('Authorization Error: make sure API Key is correctly set')
-        else:
-            raise e
-    return "ok"
+    for export_profile in export_profiles:
+        try:
+            client.export_events(export_profile)
 
+        except Exception as e:
+            if 'Forbidden' in str(e):
+                raise DemistoException('Authorization Error: Make sure client credentials are correctly set') from e
 
-def get_raw_events(client: Client, time_of_last_event: str) -> list:
+            if 'Unable to find export_profile' in str(e):
+                invalid_export_profiles.add(export_profile)
+
+            else:
+                # Some other unknown error
+                raise
+
+    if invalid_export_profiles:
+        raise DemistoException(f'Invalid export profiles: {", ".join(invalid_export_profiles)}')
+
+    return 'ok'
+
+
+def fetch_events(client: Client, export_profile: str, limit: int | None = None) -> tuple[list[dict], dict]:
     """
-       helper function that is used in get-events and fetch-events to get the actual events and sort them
-       Args:
-           client (Client): DG client
-           time_of_last_event: time of last ingested event
-       Returns:
-           list: list of events
+    Args:
+        client (Client): Digital Guardian client.
+        export_profile (str): Export profile name.
+        limit (int | None): Optional value to limit the number of yielded events.
+    Returns:
+        tuple[list[dict], dict]: Events and last run dictionary.
     """
-    outcome = []
-    event_list = []
-    if not time_of_last_event:
-        time_of_last_event_datetime = datetime.now() - timedelta(hours=1)
-        time_of_last_event_str = datetime_to_string(time_of_last_event_datetime)
-    else:
-        temp_time: datetime = arg_to_datetime(arg=time_of_last_event, required=True)  # type: ignore[assignment]
-        time_of_last_event_str = temp_time.isoformat(sep=' ', timespec='milliseconds')
-    current_time = datetime_to_string(datetime.now())
-    events = client.get_events(time_of_last_event_str, current_time)
-    events = {} if events is None else events
-    for field_names in events.get("fields"):
-        outcome.append(field_names.get('name'))
-    for event in events.get("data"):
-        result = dict(zip(outcome, event))
-        event_list.append(result)
-    event_list.sort(key=lambda item: (item.get("inc_mtime"), item.get("dg_guid")))
-    return event_list
-
-
-def get_events_command(client: Client, args: dict) -> Tuple[list, CommandResults]:
+    demisto.debug('Fetching events')
```

Comment:
```suggestion
    demisto.debug('Fetching events, {export_profile=}')
```

Reply:
Improved log message in commit de6eca13f444da175cf25459c9d3a91d521fdc81